app/v1/services/CreateAndAmendCharitableGivingTaxReliefService.scala

```diff
@@ -16,65 +16,55 @@
 
 package v1.services
 
-import cats.data.EitherT
-import javax.inject.{Inject, Singleton}
-import uk.gov.hmrc.http.HeaderCarrier
-import utils.Logging
+import api.controllers.RequestContext
+import api.models
+import api.models.errors._
+import api.services.BaseService
+import cats.implicits.toBifunctorOps
 import v1.connectors.CreateAndAmendCharitableGivingTaxReliefConnector
-import v1.controllers.EndpointLogContext
-import v1.models.errors._
 import v1.models.request.createAndAmendCharitableGivingTaxRelief.CreateAndAmendCharitableGivingTaxReliefRequest
-import v1.support.DownstreamResponseMappingSupport
 
+import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class CreateAndAmendCharitableGivingTaxReliefService @Inject() (connector: CreateAndAmendCharitableGivingTaxReliefConnector)
-    extends DownstreamResponseMappingSupport
-    with Logging {
+class CreateAndAmendCharitableGivingTaxReliefService @Inject() (connector: CreateAndAmendCharitableGivingTaxReliefConnector) extends BaseService {
 
-  def amend(request: CreateAndAmendCharitableGivingTaxReliefRequest)(implicit
-      hc: HeaderCarrier,
-      ec: ExecutionContext,
-      logContext: EndpointLogContext,
-      correlationId: String): Future[ServiceOutcome[Unit]] = {
+  def amend(
+      request: CreateAndAmendCharitableGivingTaxReliefRequest)(implicit ctx: RequestContext, ec: ExecutionContext): Future[ServiceOutcome[Unit]] = {
 
-    val result = for {
-      responseWrapper <- EitherT(connector.createAmend(request)).leftMap(mapDownstreamErrors(downstreamErrorMap))
-    } yield responseWrapper
+    connector.createAmend(request).map(_.leftMap(mapDownstreamErrors(downstreamErrorMap)))
 
-    result.value
   }
 
   private val downstreamErrorMap: Map[String, MtdError] = {
     val errors = Map(
       "INVALID_NINO"                      -> NinoFormatError,
-      "INVALID_TYPE"                      -> InternalError,
+      "INVALID_TYPE"                      -> models.errors.InternalError,
```

Comment:
Should just be InternalError without the "models.errors." (same for all of these)